### ISSUE TITLE
Idea: Add-on or plugin for Flare

### ISSUE THREAD

**Issue by sujiniku** [clintbellanger/flare-engine#1627]
About the features that have been proposed for engine but not decided to use in flare-game,  I think that these feature should be independent from the engine, such as a plug-in or add-on.

Also, flare should provide the such development environment of add-on or plugin.

**Comment by sujiniku** [comment]
Only when installing Mod, if we allow installing plugin, the player would almost prevent cracking aimed at plugins.
Although it is unknown whether it is "plug-in".

Perhaps it may be better to say "extra runtime" etc.

Also, perhaps due to software dependency at compile time, it may be technically difficult to add plugin except during Mod installation.

When updating an plugin, by asking the player to reinstall together mod and plugin, we may optimize the combination of Mod and add-on , so I think that compatibility will be easy to secure.
Also, Mod developers should provide to the players such plugin built-in Mod by himself / herself.

In other words, it is the Mod developer's responsibility and authority to incorporate plugin to use for Mod.

**Comment by sujiniku** [comment]
Perhaps in combination with the "Feature idea: Interaction in books via Events" #1625 function, 
this plugins may be written as INI files instead of C++.
We should lead the Modder to using rather only INI  as much as possible.
